examples/lband_decode/lband_decode.cc

```diff
@@ -0,0 +1,99 @@
+/**************************************************************************/ /**
+* @brief Example of decoding FusionEngine messages from a recorded file.
+* @file
+******************************************************************************/
+
+#include <cstdint>
+#include <cstdio>
+#include <fstream>
+
+#include <point_one/fusion_engine/messages/gnss_corrections.h>
+#include <point_one/fusion_engine/parsers/fusion_engine_framer.h>
+
+using namespace point_one::fusion_engine::messages;
+using namespace point_one::fusion_engine::parsers;
+
+/******************************************************************************/
+class LbandDecoder {
```

Comment:
I'd recommend removing this class entirely for clarity. It actually complicates the understanding a little bit here since now the app is creating a class and registering two different callbacks, and the reader needs to follow why. Plus, `ProcessFEInput()` registers a new callback with the framer on every read call, which is not really recommended.

Instead, I'd suggest simply defining the framer in `main()`, and then creating a callback directly to the framer that does the combined "is this an L-band message?" and "print and write to disk". Then the while loop would just call `OnData()`. `main()` would look almost exactly the same as it already does, and this class wouldn't exist at all, so the app would be much simpler to follow.

Reply:
The idea was that since this is made as an example for customer they could copy and paste the class without needing to dig into the internals. It basically just hides all the FE logic, and would make more sense as it's own file.

Reply:
Yeah that's what I figured. Makes sense, but in practice I think it actually just makes things more confusing with the multiple callbacks and stuff based on my experience with how people have used these examples in the past.